Source/WebCore/editing/MarkupAccumulator.h

```diff
@@ -127,8 +127,8 @@ class MarkupAccumulator {
 
 inline void MarkupAccumulator::endAppendingNode(const Node& node)
 {
-    if (is<Element>(node))
-        appendEndTag(m_markup, downcast<Element>(node));
+    if (RefPtr element = dynamicDowncast<Element>(node))
+        appendEndTag(m_markup, *element);
     else if (auto element = replacementElement(node))
```

Comment:
@cdumez `replacementElement()` is another method that returns RefPtr where it is not clear to me that it should per your earlier feedback. If you agree happy to tackle that in a follow-up.

Reply:
`replacementElement()` is holding a RefPtr to the element internally on the stack. Therefore there is no cost to returning a RefPtr (no extra ref'ing). Yet, if the caller needs a RefPtr, returning a (free) RefPtr is beneficial. I'd keep `replacementElement()` returning a RefPtr.

However, I would replace `auto` with `RefPtr` here for clarity. It'd be nice if you could do this before landing.

Reply:
Will do and will do the same for the other caller.